src/plugins/data/common/index_patterns/fields/field.ts

```diff
@@ -57,7 +57,7 @@ export class Field implements IFieldType {
   visualizable?: boolean;
   scripted?: boolean;
   subType?: IFieldSubType;
-  displayName?: string;
+  displayName: string;
```

Comment:
:information_source: This property was already always set via `obj.comp` so I am just adjusting the type here.